wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -432,40 +432,101 @@ def _get_es_body(self, for_count=False):
         return body
 
     def _do_search(self):
-        # Params for elasticsearch query
-        params = dict(
-            index=self.backend.get_index_for_model(self.query.queryset.model).name,
-            body=self._get_es_body(),
-            _source=False,
-            from_=self.start,
-        )
-
-        params[self.fields_param_name] = 'pk'
+        PAGE_SIZE = 100
 
-        # Add size if set
         if self.stop is not None:
-            params['size'] = self.stop - self.start
+            limit = self.stop - self.start
+        else:
+            limit = None
+
+        use_scroll = limit is None or limit > PAGE_SIZE
+
+        params = {
+            'index': self.backend.get_index_for_model(self.query.queryset.model).name,
+            'body': self._get_es_body(),
+            '_source': False,
+            'from_': self.start,
+            self.fields_param_name: 'pk',
+        }
+
+        if use_scroll:
+            params.update({
+                'scroll': '2m',
+                'size': PAGE_SIZE,
+            })
+
+            # Send to Elasticsearch
+            page = self.backend.es.search(**params)
+
+            while True:
+                if len(page['hits']['hits']) == 0:
```

Comment:
To avoid extra visual noise, define a variable like `hits = page['hits']['hits']`since it’s also used below.
Then I don’t think you have to use a length comparison as hit is a dict. Simply use `if not hit:`, it will be faster as it skips a rich comparison, a local name lookup, a function call, a tuple construct and an unoptimised length fetch.